Archipelago.MultiClient.Net.Analyzers/Archipelago.MultiClient.Net.Analyzers/Fixes/ItemFlagsInSwitchFixes.cs

```diff
@@ -0,0 +1,129 @@
+                ),
+                diagnostic
+            );
+        }
+
+        private async Task<Document> ConvertCaseToHasFlag(
+            Document document,
+            CaseSwitchLabelSyntax caseLabel,
+            CancellationToken cancellationToken)
+        {
+            DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken);
+            SwitchSectionSyntax switchSection = (SwitchSectionSyntax)caseLabel.Parent!;
+            SwitchStatementSyntax switchStatement = (SwitchStatementSyntax)switchSection.Parent!;
+            ExpressionSyntax switchExpression = switchStatement.Expression;
+            
+            string variableName = GetPatternMatchingVariableName(editor, switchStatement);
+
+            // Generate the pattern matching case label
+            CasePatternSwitchLabelSyntax newCaseLabel = GeneratePatternMatchingCaseLabel(caseLabel, switchExpression, variableName);
+
+            // Replace the old case label with the new case label
+            editor.ReplaceNode(caseLabel, newCaseLabel);
+
+            Document newDoc = editor.GetChangedDocument();
+            return newDoc;
+        }
+
+        private static string GetPatternMatchingVariableName(DocumentEditor editor, SwitchStatementSyntax switchStatement)
+        {
+            var analysis = editor.SemanticModel.AnalyzeDataFlow(switchStatement);
+
+            if (analysis is null)
+            {
+                return "f";
+            }
+            
+            var hasCollision = analysis.WrittenInside.Any(x => x.Name.StartsWith("f"));
+
+            if (hasCollision)
+            {
+                var collisions = analysis.WrittenInside.Where(x => x.Name.StartsWith("f") && int.TryParse(x.Name[1..], out var _));
+                var existingNumberedFVars = collisions.Select(x => int.Parse(x.Name[1..]));
+
+                if (!existingNumberedFVars.Any())
+                {
+                    return $"f1";
+                }
+
+                return $"f{existingNumberedFVars.Max() + 1}";
+            }
+
+            return "f";
+        }
+
+        private CasePatternSwitchLabelSyntax GeneratePatternMatchingCaseLabel(CaseSwitchLabelSyntax caseLabel, ExpressionSyntax switchExpression, string newVarName)
+        {
+            // Create the pattern matching statement: "case var f when f.HasFlag(ItemFlags.Advancement):"
+            var variableDesignation = SyntaxFactory.SingleVariableDesignation(SyntaxFactory.Identifier(newVarName));
+            var varPattern = SyntaxFactory.VarPattern(SyntaxFactory.Token(SyntaxKind.VarKeyword), variableDesignation);
+
+            var whenClause = SyntaxFactory.WhenClause(
+                SyntaxFactory.InvocationExpression(
+                    SyntaxFactory.MemberAccessExpression(
+                        SyntaxKind.SimpleMemberAccessExpression,
+                        SyntaxFactory.IdentifierName(newVarName),
+                        SyntaxFactory.IdentifierName("HasFlag")
```

Comment:
Just calling out that net35 does not have HasFlag - idk if/how we can detect the target runtime and adjust accordingly

Reply:
This is possible to detect but I will leave it out of scope for now since this problem exists elsewhere. I think #1 covers this actually